scripts/validate_sdd.py

```
@@ -433,50 +498,185 @@ def validate_invariants(sdd: str, errors: list[str]) -> int:
+            root,
+            pr_base_ref,
+            pr_head_ref,
+            "WP0 PR base-to-head",
+            expected,
+            errors,
+        )
+
+    merge_parent_ref = fields.get("wp0_merge_parent_ref", "")
+    merge_ref = fields.get("wp0_merge_ref", "")
+    if pr_base_check.returncode == 0:
+        merge_ancestry = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)